packages/edit-site/src/components/navigation-sidebar/navigation-panel/new-template-dropdown.js

```diff
@@ -50,7 +50,7 @@ export default function NewTemplateDropdown() {
 			excerpt: description,
 			// Slugs need to be strings, so this is for template `404`
 			slug: slug.toString(),
-			status: 'draft',
+			status: 'publish',
```

Comment:
This change is not related to the fix, but it was something I've tried changing when investigating.
Thing is, templates are always created as `publish`, so using `draft` here would only be confusing.
There are other mentions of draft templates in the Navigation sidebar (e.g. drafts items would have a `[Draft]` label prepended to the title.
I haven't cleaned them up for now, as I don't think they'd be as confusing as this.